src/transport/client.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{
+    common::Endpoint,
+    error::{Error, Result},
+    transport::{proto::RequestKind, Request, Response},
+};
+use tokio_sync::{mpsc, oneshot};
+
+pub type Channel = mpsc::Sender<(Request, oneshot::Sender<crate::Result<Response>>)>;
+
+#[derive(Debug, Clone)]
+pub struct Client {
+    inner: Channel,
+}
+
+impl Client {
```

Comment:
~Something else to think about is the fact that there is one use case for the client where a target is specified. It's when [you're handing a Phase1aMessage](https://github.com/lalithsuresh/rapid/blob/master/rapid/src/main/java/com/vrg/rapid/Paxos.java#L131). Though the remote being set here is the node from which the request was received in the first place.~

Just saw the `.send` impl here. Lol, my bad.